test/test_bugFixes.js

```diff
@@ -0,0 +1,40 @@
+const assert = require('assert');
+const grammar = require('../js/main.js');
+
+
+describe('Test bug fixes', () => {
+  beforeEach(function() {
+    if (global.gc) { global.gc(); }
+  });
+
+  it('text with "id XXX" should not capitalize XXX', () => {
```

Comment:
For each of these test cases, it would be good if you could add the link to the related GitHub issue as a comment.

Reply:
Done. Also made a few more changes.